asyncscope.md

```diff
@@ -1054,130 +1095,186 @@ struct counting_scope {
+
+[*] --> unused
+unused --> open : nest()
+unused --> unused_and_closed : close()
+unused --> open_and_joining : join-sender\nstarted
+open --> closed : close()
+open --> open_and_joining : join-sender\nstarted
+closed --> closed_and_joining : join-sender\nstarted
+open_and_joining --> closed_and_joining : close()
+unused_and_closed --> closed_and_joining : join-sender\nstarted
+closed_and_joining --> joined : count reaches 0\njoin-sender completes
+open_and_joining --> joined : count reaches 0\njoin-sender completes
+joined --> [*] : \~simple_counting_scope()
+unused_and_closed --> [*] : \~simple_counting_scope()
+unused --> [*] : \~simple_counting_scope()
+
+@enduml
+```
 
-[comment]: # (TODO: The following paragraph is much less convincing after splitting scopes into scopes and tokens.)
+_Note: a scope is "open" if its current state is unused, open, or open-and-joining; a scope is "closed" if its current
+state is closed, unused-and-closed, closed-and-joining, or joined._
+
+Instances start in the unused state after being constructed. This is the only time the scope's state can be set to
+unused. When the `simple_counting_scope` destructor starts, the scope must be in the unused, unused-and-closed, or
+joined state; otherwise, the destructor invokes `std::terminate()`. Permitting destruction when the scope is in the
+unused or unused-and-closed state ensures that instances of `simple_counting_scope` can be used safely as data-members
+while preserving structured functionality.
+
+Connecting and starting a _`join-sender`_ returned from `join()` moves the scope to either the open-and-joining or
+closed-and-joining state. Merely calling `join()` or connecting the _`join-sender`_ does not change the scope's
+state---the _`operation-state`_ must be started to effect the state change. A started _`join-sender`_ completes when the
+scope's count of outstanding operations reaches zero, at which point the scope transitions to the joined state.
+
+Calling `close()` on a `simple_counting_scope` moves the scope to the closed, unused-and-closed, or closed-and-joining
+state, and causes all future calls to `nest()` to fail.
+
+Any call to `nest()` may throw an exception if copying or moving the input sender into the returned _`nest-sender`_
+throws an exception. `nest()` provides the Strong Exception Guarantee so the scope's state is left unchanged if an
+exception is thrown while constructing the returned _`nest-sender`_. Assuming `nest()` does not throw:
+
+- While a scope is in the unused, open, or open-and-joining state, calls to `nest()` succeed by returning an "associated
+  sender" (see below) and incrementing the scope's count of outstanding operations _before returning_.
+- While a scope is in the closed, unused-and-closed, closed-and-joining, or joined state, calls to `nest()` fail by
+  returning an "unassociated sender" (see below). Failed calls to `nest()` do _not_ increment the scope's count of
+  outstanding operations.
```

Comment:
```suggestion
exception is thrown while constructing the returned _`nest-sender`_. 

Assuming `nest()` does not throw:
- While a scope is in the unused, open, or open-and-joining state, calls to `nest()` succeed by returning an "associated
  sender" (see below) and incrementing the scope's count of outstanding operations _before returning_.
- While a scope is in the closed, unused-and-closed, closed-and-joining, or joined state, calls to `nest()` fail by
  returning an "unassociated sender" (see below). Failed calls to `nest()` do _not_ increment the scope's count of
  outstanding operations.
```

Reply:
The only change you're suggesting here is moving "Assuming nest() does not throw:" to its own line, right?

Reply:
Yup